pkg/proxy/iptables/proxier.go

```diff
@@ -1619,7 +1633,7 @@ func (proxier *Proxier) syncProxyRules() {
 
 	// Close old local ports and save new ones.
 	for k, v := range proxier.portsMap {
-		if replacementPortsMap[k] == nil {
+		if v != nil && replacementPortsMap[k] == nil {
```

Comment:
was this causing a panic?

Reply:
Yes, if add the `nodePort` in svc spec in that unit test.

Reply:
Tried adding `nodePort` in `TestEndpointSliceE2E`, same panic.

Reply:
that makes sense, can you swap the check:
```
if replacementPortsMap[k] == nil && v != nil {
```

and put this change in a separate commit

Reply:
Oh fixed. Put it as the first commit and rebased the changes on top of that.